### DISCUSSION TITLE
Default value for max_distance searching for matches

### DISCUSSION THREAD

**Discussion by MiguelGarcaoSilva** [TDAmeritrade/stumpy#931] [Q&A]
Hello,

I've noticed the default value for the **max_distance** parameter in the **stumpy.motifs** function is set to _np.nanmax([np.nanmean(D) - 2.0 * np.nanstd(D), np.nanmin(D)])_ . Is there a specific reason to use this value as default? Are there any published papers recommending it? 

Thanks in advance!

**Comment by seanlaw** [comment]
@MiguelGarcaoSilva Thank you for your question and welcome to the STUMPY community!

> I've noticed the default value for the max_distance parameter in the stumpy.motifs function is set to np.nanmax([np.nanmean(D) - 2.0 * np.nanstd(D), np.nanmin(D)]) . Is there a specific reason to use this value as default? Are there any published papers recommending it?

To be honest, this is just a super crude/naive way to identify motif distances that are (possibly) exceptional compared with all other distances. We (naively) assume that distribution of matrix profile distances follow a normal distribution and that all distances below 2 stddevs from the mean are "interesting"/"significant" to look at first. Since the time series can vary drastically from use case to use case, this naive default value adjusts according to the time series data being used.

Also, choosing the `nanmax` between the "2 stddevs from the mean" and `nanmin` ensures that you don't get a `max_distance` that is below the lowest possible matrix profile distance computed (since it is completely possible that, say, 2 stddevs below the mean could be a negative value, which would make no sense).

Note that this is a sane/reasonable starting point but you are free to specify your own `max_distance` based on your domain expertise. I hope that helps.
